### ISSUE TITLE
`tuist test` fails due to codesigning xcargs

### ISSUE THREAD

**Issue by hisaac** [tuist/tuist#3920]
I've been having issues running some specific unit tests in my project that require a test host app. After some investigation, I think I've nailed down what the issue is.

The tests run fine when I run them through Xcode, and when I run them manually using `xcodebuild` in the terminal, but when I use `tuist test` to run the tests, the test host app crashes, and the unit tests fail. `tuist test` is just running an `xcodebuild` command itself in order to run the tests, so I was able to compare the command I was using to the one Tuist generates, and figure out what's causing the issue.

This is the command run by `tuist test`:

```shell
/usr/bin/xcrun \
	xcodebuild \
	test \
	-scheme GSPAuthTests \
	-workspace /Users/Z006TJ7/.tuist/Cache/Projects/32900f8a55d0a971b07cdaad75565aa4/Tuist_Flagship.xcworkspace \
	-sdk iphonesimulator \
	-destination id=0D4CFB5D-84F5-4ED4-8FF2-421590279D6A \
	CODE_SIGN_IDENTITY= \
	CODE_SIGNING_REQUIRED=NO \
	CODE_SIGN_ENTITLEMENTS= \
	CODE_SIGNING_ALLOWED=NO
```

Those last four `CODE_SIGN*` arguments are the ones causing the tests to fail. When I omit those arguments, the test run without issue.

In [`TestService.swift`](https://github.com/tuist/tuist/blob/754bdf0a672a0080cb20071b4a332fa2dd5884ca/Sources/TuistKit/Services/TestService.swift#L225), Tuist is explicitly saying to skip codesigning, which is what causes those arguments to be added to the command.

```swift
_ = try xcodebuildController.test(
    .workspace(graphTraverser.workspace.xcWorkspacePath),
    scheme: scheme.name,
    clean: clean,
    destination: destination,
    derivedDataPath: nil,
    resultBundlePath: resultBundlePath,
    arguments: buildGraphInspector.buildArguments(
        project: buildableTarget.project,
        target: buildableTarget.target,
        configuration: configuration,
        skipSigning: true <---------------------------------
    )
)
```

If I switch that argument to `false`, the generated `xcodebuild` command omits the `CODE_SIGN*` arguments, and the tests run without issue.

I'm not familiar enough to know _why_ this would cause a failure, or why this is hardcoded to skip signing. Is there some way to determine when signing will be required? Is there something in my test target I can change that would address this?

## Steps To Reproduce

I don't really know the specific reasons why this test host app fails due to the signing issues, so I don't have specific steps to reproduce right now. If necessary, I can try to work up an example project.

## Expected Behavior

Tests run successfully.

## Environment

- macOS: 12.1
- Tuist: 2.5.0

## Additional Context

These specific unit tests require a test host, because they are testing Keychain functionality. I'm not sure if that is related to whether or not codesigning is required, but thought it would be worth mentioning.

**Comment by hisaac** [comment]
Small update: Looks like the actual argument that's causing this issue is `CODE_SIGNING_ALLOWED=NO`. If I remove just that one, but still pass the others, things work fine.

I wonder, are these settings passed onto the test host somehow? I think perhaps my test hose _does_ need code signing because it requires an entitlements file for the keychain functionality, so perhaps disallowing code signing for the test target is also disabling it for the test host, and causing the test host to crash.

**Comment by hisaac** [comment]
I was able to reproduce this in an example project: [CodeSignExample.zip](https://github.com/tuist/tuist/files/7790581/CodeSignExample.zip)

This project has some tests that require access to a device's Keychain, and as such, require a test host app to be running.

To test this, from the root of the project, run the following shell command:

```shell
xcodebuild \
      test \
      -project CodeSignExample.xcodeproj \
      -scheme CodeSignExample \
      -destination 'platform=iOS Simulator,name=iPhone 13,OS=15.2' \
      CODE_SIGN_IDENTITY= \
      CODE_SIGNING_REQUIRED=NO \
      CODE_SIGN_ENTITLEMENTS=
```

Note that the app is launched in the simulator, and the tests pass.

Next, run the same command, but add the `CODE_SIGNING_ALLOWED=NO` flag:

```shell
xcodebuild \
      test \
      -project CodeSignExample.xcodeproj \
      -scheme CodeSignExample \
      -destination 'platform=iOS Simulator,name=iPhone 13,OS=15.2' \
      CODE_SIGN_IDENTITY= \
      CODE_SIGNING_REQUIRED=NO \
      CODE_SIGN_ENTITLEMENTS= \
      CODE_SIGNING_ALLOWED=NO
```

Note that the app launches, but then crashes, and the tests fail.

I'm still not sure _exactly_ what's going on, but my hunch is like I said above: explicitly disallowing codesigning from the command line passes the behavior to the test host as well, thereby breaking any tests that have a test host that requires codesigning to function.

And so, I'd like to propose that we remove the `CODE_SIGNING_ALLOWED=NO` flag from the `xcodebuild test` invocation used by Tuist. I've got a PR open with those changes here: #3924
